### DISCUSSION TITLE
Abuse signed url from storage

### DISCUSSION THREAD

**Discussion by Vinzent03** [supabase/supabase#3564] [Questions] [answered]
Let's say I have a file that a user can decide about whether it should be public or private. The file is stored in a private bucket. The file is set to public and someone generates a signed url to get access to the file with a duration of 5 years. Now the user sets the file to private. So no one but himself can access the file. But the one with the signed url has still access right? And the only way to remove the access is to either rename the file or to reset the jwt secret, right? 

I know this is really theoretically, but I want to understand the system behind it.

**Comment by dshukertjr** [comment]
Hi @Vinzent03 !

Supabase is using Amazon S3 behind the scenes, so you can learn more about how the signed URL works by reading articles like [this](https://advancedweb.hu/how-s3-signed-urls-work/) that talk about signed URLs in S3!

**Comment by Vinzent03** [comment, marked as answer]
Thanks for the article. The only solution I can think of is to deny access to anon users and to write an own api endpoint to generate a jwt with a short expiration time to the anon users to prevent them to generate jwt with a long expiration time.

**Comment by fachneubw** [comment]
no canteri

در تاریخ یکشنبه ۱۷ اکتبر ۲۰۲۱،‏ ۲:۱۸ Vinzent ***@***.***>
نوشت:
